tests/tuning/hyperparams/test_base.py

```diff
@@ -0,0 +1,188 @@
+        # setup
+        values = [1]
+        dimension = 2
+
+        # run
+        with self.assertRaises(ValueError):
+            to_array(values, dimension)
+
+    def test_to_array_list_values_gt_dimension(self):
+        # setup
+        values = [1, 2, 3]
+        dimension = 2
+
+        # run
+        with self.assertRaises(ValueError):
+            to_array(values, dimension)
+
+    def test_to_array_list_values_eq_dimension(self):
+        # setup
+        values = [1, 2, 3]
+        dimension = 3
+
+        # run
+        result = to_array(values, dimension)
+
+        # assert
+        expected_result = np.array([[1, 2, 3]])
+        np.testing.assert_array_equal(result, expected_result)
+
+    def test_to_array_invalid_values_shape(self):
+        # setup
+        values = [[[1]]]
+        dimension = 1
+
+        # run
+        with self.assertRaises(ValueError):
+            to_array(values, dimension)
+
+    def test_to_array_invalid_values_dimension(self):
+        # setup
+        values = [[1, 2], [1, 2, 3]]
+        dimension = 2
+
+        # run
+        with self.assertRaises(ValueError):
+            to_array(values, dimension)
+
+
+class TestBaseHyperParam(TestCase):
+
+    class TestHyperParam(BaseHyperParam):
+
+        K = 1
+
+        def _inverse_transform(self, values):
+            pass
+
+        def _transform(self, values):
+            pass
+
+        def sample(self, values):
+            pass
+
+    def test__within_range_valid(self):
+        # setup
+        instance = self.TestHyperParam()
+        values = np.array([[1], [2]])
+        _min = 0
+        _max = 10
+
+        # run / assert
+        instance._within_range(values, min=_min, max=_max)
+
+    def test__within_range_invalid(self):
+        # setup
+        instance = self.TestHyperParam()
+        values = np.array([[1], [2]])
+        _min = 0
+        _max = 1
+
+        # run / assert
+        with self.assertRaises(ValueError):
+            instance._within_range(values, min=_min, max=_max)
+
+    @patch('btb.tuning.hyperparams.base.BaseHyperParam._within_range')
+    def test__within_search_space(self, mock__within_range):
+        # setup
+        instance = self.TestHyperParam()
+        values = 15
+
+        # run
+        instance._within_search_space(values)
+
+        # assert
+        mock__within_range.assert_called_once_with(values, min=0, max=1)
+
+    @patch('btb.tuning.hyperparams.base.BaseHyperParam._within_range')
+    def test__within_hyperparam_space(self, mock__within_range):
+        # setup
+        instance = self.TestHyperParam()
+        instance.min = 0
+        instance.max = 10
+        values = 5
+
+        # run
+        instance._within_hyperparam_space(values)
+
+        # assert
+        mock__within_range.assert_called_once_with(values, min=0, max=10)
+
+    @patch('btb.tuning.hyperparams.base.BaseHyperParam._within_search_space')
+    @patch('btb.tuning.hyperparams.base.to_array')
```

Comment:
Let's try to use less mocks in general. Only use mocks when:
- You need access to external resources
- You interact with randomness that needs to be controlled
- You are calling other methods that are slow or complex, or that need mocks to run.

In this case, you can simply:
- Pass real values
- let `to_array` transform the values
- let `_within_range` validate the values
- finally check that _inverse_transform gets the expected values using a mock.